tests/background/hypothesis-chrome-extension-test.js

```diff
@@ -268,11 +268,56 @@ describe('HypothesisChromeExtension', function () {
       it('resets the tab state when loading', function () {
         const tab = createTab({
           state: 'active',
+          annotationCount: 8,
           ready: true,
           extensionSidebarInstalled: true,
         });
         fakeChromeTabs.onUpdated.listener(tab.id, { status: 'loading' }, tab);
         assert.equal(tabState[tab.id].ready, false);
+        assert.equal(tabState[tab.id].annotationCount, 0);
+        assert.equal(tabState[tab.id].extensionSidebarInstalled, false);
+      });
+
+      it('ignores consecutive `loading` events for the same URL and tab until the loading is completed', function () {
+        const tab = createTab({
+          state: 'active',
+          annotationCount: 8,
+          ready: true,
+          extensionSidebarInstalled: true,
+        });
+        fakeChromeTabs.onUpdated.listener(tab.id, { status: 'loading' }, tab);
+        assert.equal(tabState[tab.id].ready, false);
+        assert.equal(tabState[tab.id].annotationCount, 0);
+        assert.equal(tabState[tab.id].extensionSidebarInstalled, false);
+
+        tabState[tab.id].annotationCount = 5;
+        fakeChromeTabs.onUpdated.listener(tab.id, { status: 'loading' }, tab); // ignored
+        assert.equal(tabState[tab.id].annotationCount, 5);
+
+        fakeChromeTabs.onUpdated.listener(tab.id, { status: 'complete' }, tab);
+        fakeChromeTabs.onUpdated.listener(tab.id, { status: 'loading' }, tab);
+        assert.equal(tabState[tab.id].ready, false);
+        assert.equal(tabState[tab.id].annotationCount, 0);
+        assert.equal(tabState[tab.id].extensionSidebarInstalled, false);
+      });
+
+      it('resets the tab state when loading a different URL (even when previous loading event did not completed)', function () {
+        const tab = createTab({
+          state: 'active',
+          annotationCount: 8,
+          ready: true,
+          extensionSidebarInstalled: true,
+        });
+        fakeChromeTabs.onUpdated.listener(tab.id, { status: 'loading' }, tab);
+        assert.equal(tabState[tab.id].ready, false);
+        assert.equal(tabState[tab.id].annotationCount, 0);
+        assert.equal(tabState[tab.id].extensionSidebarInstalled, false);
+
+        tabState[tab.id].annotationCount = 5;
+        tab.url += '#new-fragment';
```

Comment:
If only the URL fragment has changed, we probably _shouldn't_ reset the badge counter, since that will never affect the result. I don't think we need to address that in this PR, but I suggest to use a completely different URL in this test to illustrate the expected behavior.

Reply:
It does indeed the case. Thought the code is executed when the fragment is changed, the code that fetches the badge request is not (within a certain period). 

This is because there is a normalisation in the URL and the fragment is removed. See [here](https://github.com/hypothesis/browser-extension/blob/13d58001449c65c9341379118e4264367855b8a7/tests/background/uri-info-test.js#L49-L67).